src/main/java/org/jenkinsci/plugins/authorizeproject/strategy/SystemAuthorizationStrategy.java

```diff
@@ -0,0 +1,227 @@
+     */
+    @Override
+    public final boolean equals(Object obj) {
+        return obj != null && SystemAuthorizationStrategy.class == obj.getClass();
+    }
+
+    /**
+     * Our descriptor
+     */
+    @Extension(ordinal = -100)
+    public static class DescriptorImpl extends AuthorizeProjectStrategyDescriptor {
+
+        /**
+         * Flag to mark where changing a job using this strategy requires administrator permissions.
+         */
+        private boolean permitReconfiguration;
+
+        /**
+         * @return the name shown in project configuration pages.
+         * @see hudson.model.Descriptor#getDisplayName()
+         */
+        @Override
+        public String getDisplayName() {
+            return Messages.SystemAuthorizationStrategy_DisplayName();
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean isEnabledByDefault() {
+            return false;
+        }
+
+        /**
+         * Gets the flag to mark where changing a job using this strategy requires administrator permissions.
+         *
+         * @return {@code true} if non-admins are allowed to modify jobs that are using this strategy.
+         */
+        public boolean isPermitReconfiguration() {
+            return permitReconfiguration;
+        }
+
+        /**
+         * Sets the flag to mark where changing a job using this strategy requires administrator permissions.
+         *
+         * @param permitReconfiguration {@code true} if non-admins are allowed to modify jobs that are using this strategy.
+         */
+        public void setPermitReconfiguration(boolean permitReconfiguration) {
+            if (permitReconfiguration != this.permitReconfiguration) {
+                this.permitReconfiguration = permitReconfiguration;
+                save();
+            }
+        }
+
+        /**
+         * Tests if an object is a {@link Job}
+         *
+         * @param it the object.
+         * @return {@code true} if and only if the supplied object is a {@link Job}
+         */
+        @Restricted(NoExternalUse.class) // helper method for Jelly EL
+        public boolean isJob(Object it) {
+            return it instanceof Job;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void configureFromGlobalSecurity(StaplerRequest req, JSONObject js) throws FormException {
+            setPermitReconfiguration(js.getBoolean("permitReconfiguration"));
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public SystemAuthorizationStrategy newInstance(StaplerRequest req, JSONObject formData)
+                throws FormException {
+            SystemAuthorizationStrategy result = (SystemAuthorizationStrategy) super.newInstance(req, formData);
+            Jenkins instance = Jenkins.getInstance();
+            if (instance == null || !instance.hasPermission(Jenkins.RUN_SCRIPTS)) {
```

Comment:
`instance == null` ?   = yet still try to go on and find a job - seems wonky to me?